src/value.c

```diff
@@ -376,12 +394,36 @@ void SIValue_Print(FILE *outstream, SIValue *v) {
   }
 }
 
-void SIValue_Free(SIValue *v) {
-	// Only values of type T_STRING are heap allocations owned by the SIValue
-	// that need to be freed.
-	if (v->type == T_STRING) {
-		rm_free(v->stringval);
-		v->stringval = NULL;
+void SIValue_Persist(SIValue *v) {
+	if (v->allocation == M_VOLATILE) {
+		if (v->type == T_NODE) {
```

Comment:
Unfortunately, putting string handling here can cause some conflicts in allocation management.

For example, in the query:
`MATCH (Cameron:actor{name:"Cameron Diaz"})-[:act]->(m:movie) RETURN Cameron.name, COUNT(m.title)`

A record will hold a string scalar for "Cameron Diaz" with `M_CONST` ownership (it's allocated in the datablock). That scalar becomes a group key for op_aggregate. If we update the logic here to move the string's ownership, it will get freed by the `Record_Free` call in the AggregateConsume, and then invalidly accessed as a member of the group keys.